jsettlers.logic/src/main/java/jsettlers/logic/map/grid/MainGrid.java

```diff
@@ -1498,6 +1517,33 @@ public final void pushMaterialsTo(ShortPoint2D position, EMaterialType type, byt
 		}
 
 		@Override
+		public void setDock(int[] position, boolean place, Player player) {
+			if (place) { // place dock
+				int x = position[0];
+				int y = position[1];
+				short partition = landscapeGrid.getBlockedPartitionAt(x - position[2], y - position[3]);
+				for (int i = 0; i < 3; i++) {
+					ShortPoint2D point = new ShortPoint2D(x, y);
```

Comment:
if you convert `position` to an object, give it a `getPoint(int offset)` method ;-).